psi4/src/psi4/libfock/cubature.cc

```diff
@@ -3070,19 +3070,22 @@ OrientationMgr::LMatrix OrientationMgr::RotMatrixFromOneAxis(LVector axis3)
     return Q;
 }
 
+#define EPSILON 1e-14
 // I hate to write wrapper functions, but it's easier this way...
 void OrientationMgr::diagonalize(LMatrix const& M, LMatrix *Q_out, LVector *D_out)
 {
     // This is the easiest way to get the eigensystem right now.
     // The eigenvalues are returned in ascending order.
     Matrix I("Matrix to be diagonalized", 3, 3);
     double** Ip = I.pointer();
-    Ip[0][0]            = M.xx;
-    Ip[1][1]            = M.yy;
-    Ip[2][2]            = M.zz;
-    Ip[1][0] = Ip[0][1] = M.xy;
-    Ip[2][0] = Ip[0][2] = M.xz;
-    Ip[2][1] = Ip[1][2] = M.yz;
+    // Clean matrix by setting tiny values arising from numerical errors to zero
+    // to prevent generation of weird eigenvectors
+    Ip[0][0]            = (fabs(M.xx) < EPSILON ? 0.0 : M.xx);
```

Comment:
I dont think `EPSILON` will be used anywhere else. Can you just init epsilon as a double when you start setting `Ip`?